src/System.Private.CoreLib/shared/System/Globalization/CalendarData.Unix.cs

```diff
@@ -134,16 +134,103 @@ private static bool EnumDatePatterns(string localeName, CalendarId calendarId, C
             {
                 List<string> datePatternsList = callbackContext.Results;
 
-                datePatterns = new string[datePatternsList.Count];
                 for (int i = 0; i < datePatternsList.Count; i++)
                 {
-                    datePatterns[i] = NormalizeDatePattern(datePatternsList[i]);
+                    datePatternsList[i] = NormalizeDatePattern(datePatternsList[i]);
                 }
+
+                if (dataType == CalendarDataType.ShortDates)
+                    FixDefaultShortDatePattern(datePatternsList);
+
+                datePatterns = datePatternsList.ToArray();
             }
 
             return result;
         }
 
+        // FixDefaultShortDatePattern will convert the default short date pattern from using 'yy' to using 'yyyy'
+        // And will ensure the original pattern still exist in the list
+        private static void FixDefaultShortDatePattern(List<string> shortDatePatterns)
+        {
+            if (shortDatePatterns.Count == 0)
+                return;
+
+            string s = shortDatePatterns[0];
+
+            if (s.Length > 100)
+                return;
+
+            Span<char> modifiedPattern = stackalloc char[s.Length + 2];
+            int index = 0;
+
+            while (index < s.Length)
+            {
```

Comment:
Why are ticks getting special treatment? A comment with some example inputs and outputs might help.

Reply:
I am not sure what you mean by ticks here. I thought the comment before the method should be clear. I'll try to clarify it. basically, if try to format any date with the short pattern on Linux the year will be displayed as 2 digits. "6/5/18" this formatting has nothing to do with the date ticks. the fix is to display the year as 4 digits